### PR TITLE
build wheels with github-actions

### PR DESCRIPTION
**Pull request by hiaselhans**
This adds a github-actions workflow to build the wheels in all major python versions on all major platforms.

to take it one step further, this could be a deployment step but i don't want to push it on you:

``` yml
  deploy:
    runs-on: ubuntu-latest
    needs: [build-wheel-linux, build-wheel-matrix]
    steps:
      - name: checkout source code
        uses: actions/checkout@v1
      - name: Load artifacts
        uses: actions/download-artifact@v2
        with:
          name: dist
          path: dist
      - name: Remove linux_x86_64 wheels
        run: rm dist/*-linux_x86_64.whl
      - name: Publish distribution package to PyPI
        uses: pypa/gh-action-pypi-publish@master
        with:
          user: ${{ secrets.pypi_user }}
          password: ${{ secrets.pypi_password }}
```

### CODE REVIEW

**Comment by inducer** [comment]
Thanks for proposing this. I'm OK with the deployment step being added.

**Comment by hiaselhans** [comment]
Thanks @inducer 

I added a deployment task that runs on published releases
For it to work you need to add a secret to the github-repo:

PYPI_API_TOKEN: https://pypi.org/help/#apitoken

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+jobs:
+  build-wheel-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+      - uses: RalfG/python-wheels-manylinux-build@v0.3.4-manylinux2014_x86_64
+        with:
+          python-versions: 'cp37-cp37m cp38-cp38 cp39-cp39 cp310-cp310 cp311-cp311'
```
[the remark this answers is not in the document]
Comment (hiaselhans) [reply .github/workflows/deploy.yml:16]: that looks nice, but i haven't used it and don't know in which timeframe i can update the PR.
Can we use it as-is and i will send a new PR when i have tested cibuildwheel (2-3 months?)

[the remark this answers is not in the document]
Comment (hiaselhans) [reply .github/workflows/deploy.yml]: Turns out, cibuildwheel is a fairly straightforward tool and besides long build times is great to use.
The PR is updated and I hope the deploy step also works without issues.

Thanks for the hint!

**Comment by hiaselhans** [comment]
@inducer 
Should we try it?

**Comment by inducer** [comment]
Thanks, looks good to me. I'll roll a release, to see how it goes.

**Comment by inducer** [comment]
https://github.com/inducer/meshpy/releases/tag/v2022.1

**Pull request merged**